### DISCUSSION TITLE
external input & built-in mic in aggregate device with blackhole 2ch not working when mic disconnected (works on 16 ch)

### DISCUSSION THREAD

**Discussion by ghost** [ExistentialAudio/BlackHole#743] [Q&A]
So I set up an aggregate device with blackhole 16 ch followed by the built-in mic and speakers and an external mic & headphones (just a simple headset really). Works like a charm both when the headset is connected and when it isn't.

Now I don't need 16 channels any more so uninstalled that version and installed the 2 ch version. Set up the aggregate device the same way and it works fine so long as the headset is connected, but as soon as it's not, suddenly there's no input coming from the aggregate device's port assigned to the built-in mic either. In Ableton I can see that there's no signal there. Re-connect the headset, and boom, I can see signal on both the aggregate device's port assigned to the headset AND the one assigned to the built-in mic.

Again, not happening with the 16 ch version. When I disconnect the headset, Ableton still reports a signal coming from the aggregate device's port assigned to the built-in mic just fine, only the one associated with the (now not present) headset mic has gone silent as it should.

Wondering why that could be and how to solve the issue. Happy to provide screenshots/videos/more details

**Comment by devinroth** [comment]
Screenshots / video would be helpful.

Devin

**Reply by ghost** [reply]
Hey hanks for the answer,

you got it, here's a 2 min video where I recreate the issue:

https://drive.google.com/file/d/1itSTSjgiSdbTI48eI_BIiWM3AZmatE4Q/view?usp=sharing

**Comment by devinroth** [comment]
That’s crazy odd.

What happens if you have an aggregate device without BlackHole?

Devin

**Reply by ghost** [reply]
I tried with the 16 ch version first and then with the 2 ch version installed, just to be sure. When removing either one from the aggregate device, the inputs work just fine, i.e. when the headset is disconnected, the aggregate device channel associated with the built-in mic shows signal, the one associated with the (now unplugged) external mic does not.

**Comment by devinroth** [comment]
When you unplug, what errors do you get in Console from searching from searching coreaudiod?

The driver code for BlackHole 16ch and BlackHole 2ch is exactly the same. It’s most likely an issue with aggregate devices. Not much I can do about that unfortunately.

Devin

**Reply by ghost** [reply]
No messages at all in any of the left-side submenus.

I can just stick with the 16 channel version. A little extra flexibility never hurt anyone 🙂

Thanks for looking into this and just in general for the software!
